src/app/src/main/kotlin/org/imperial/mrc/hint/security/SecurityConfig.kt

```diff
@@ -79,6 +68,17 @@ class Session(
         }) as CommonProfile
     }
 
+    fun getAccessToken(): String
+    {
+        val manager = ProfileManager(webContext, sessionStore)
+
+        val profile = manager.profiles.singleOrNull() ?: OAuth20Profile()
+
+        val token = profile.getAttribute(ACCESS_TOKEN) ?: ""
```

Comment:
How does the access token get into the profile? I can't see anywhere that happens - it is handled automatically by Pac4j?

Should be throwing an error here rather then returning empty string if there's no token in the profile? I think we should only be requesting it here it if we're expecting it to exist?

Reply:
For some reason it might not exists, perhaps that's why PAC4j makes it optional. I do not think we should throw an error, authentication should be managed dynamically. For example, if a request is made to retrieve resources without access_token it should return only the public datasets and the app should still be usable.

Reply:
That would indicate that something was wrong though, wouldn't it? If we're configured to support SSO and the user is accessing ADR (which should be done by authenticated users only) then there's definitely a problem if there's no access token I think? I guess we can still give access to public datasets, but maybe we should at least log that there's an auth issue rather than just silently passing the empty token.